### PR TITLE
fix: Don't crash language servers if telemetry cannot be initialized

### LINKED ISSUE #1384: Don't crash language servers if telemetry cannot be initialized
If we have bad version from telemetry it crashes language servers.

Here a sampel of stack trace:

```
Error while consuming LSP 'textDocument/semanticTokens/full' request

java.util.concurrent.ExecutionException: com.redhat.devtools.lsp4ij.server.CannotStartServerException: Error while starting language server 'qute' (pid=null)
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at com.redhat.devtools.lsp4ij.internal.CompletableFutures.waitUntilDone(CompletableFutures.java:119)
	at com.redhat.devtools.lsp4ij.features.semanticTokens.LSPSemanticTokensHighlightVisitor.visit(LSPSemanticTokensHighlightVisitor.java:92)
	at com.intellij.codeInsight.daemon.impl.InjectedGeneralHighlightingPass.lambda$runHighlightVisitorsForInjected$9(InjectedGeneralHighlightingPass.java:311)
	at com.redhat.devtools.lsp4ij.features.semanticTokens.LSPSemanticTokensHighlightVisitor.analyze(LSPSemanticTokensHighlightVisitor.java:66)
	at com.intellij.codeInsight.daemon.impl.InjectedGeneralHighlightingPass.runHighlightVisitorsForInjected(InjectedGeneralHighlightingPass.java:308)
	at com.intellij.codeInsight.daemon.impl.InjectedGeneralHighlightingPass.lambda$addInjectedPsiHighlights$7(InjectedGeneralHighlightingPass.java:246)
	at com.intellij.codeInsight.daemon.impl.AnnotatorRunner.lambda$runAnnotatorsAsync$2(AnnotatorRunner.java:75)
	at com.intellij.concurrency.JobLauncherImpl.procInOrderAsync(JobLauncherImpl.java:537)
	at com.intellij.codeInsight.daemon.impl.AnnotatorRunner.runAnnotatorsAsync(AnnotatorRunner.java:70)
	at com.intellij.codeInsight.daemon.impl.InjectedGeneralHighlightingPass.addInjectedPsiHighlights(InjectedGeneralHighlightingPass.java:245)
	at com.intellij.codeInsight.daemon.impl.InjectedGeneralHighlightingPass.lambda$collectInformationWithProgress$3(InjectedGeneralHighlightingPass.java:86)
	at com.intellij.codeInsight.daemon.impl.InjectedGeneralHighlightingPass.lambda$processInjectedPsiFiles$5(InjectedGeneralHighlightingPass.java:184)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.probeElementsUpInner(InjectedLanguageUtilBase.java:251)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.lambda$probeElementsUp$0(InjectedLanguageUtilBase.java:217)
	at com.intellij.openapi.application.impl.ReadActionCacheImpl$allowInWriteAction$1.invoke(ReadActionCacheImpl.kt:41)
	at com.intellij.openapi.application.impl.ReadActionCacheImpl$allowInWriteAction$1.invoke(ReadActionCacheImpl.kt:41)
	at com.intellij.openapi.application.impl.ReadActionCacheImpl.allowInWriteAction(ReadActionCacheImpl.kt:29)
	at com.intellij.openapi.application.impl.ReadActionCacheImpl.allowInWriteAction(ReadActionCacheImpl.kt:41)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.probeElementsUp(InjectedLanguageUtilBase.java:216)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtilBase.enumerate(InjectedLanguageUtilBase.java:159)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.enumerateEx(InjectedLanguageManagerImpl.java:379)
	at com.intellij.codeInsight.daemon.impl.InjectedGeneralHighlightingPass.lambda$processInjectedPsiFiles$6(InjectedGeneralHighlightingPass.java:182)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:139)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:152)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:152)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:152)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:152)
	at com.intellij.openapi.application.impl.RwLockHolder.tryRunReadAction(RwLockHolder.kt:310)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:958)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$1(ApplierCompleter.java:96)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:660)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:735)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:691)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:659)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:108)
	at com.intellij.concurrency.ApplierCompleter.lambda$compute$0(ApplierCompleter.java:86)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:199)
	at com.intellij.openapi.application.impl.RwLockHolder.executeByImpatientReader(RwLockHolder.kt:491)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:86)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: com.redhat.devtools.lsp4ij.server.CannotStartServerException: Error while starting language server 'qute' (pid=null)
	at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$11(LanguageServerWrapper.java:316)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:990)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:974)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.helpAsyncBlocker(ForkJoinPool.java:1253)
	at java.base/java.util.concurrent.ForkJoinPool.helpAsyncBlocker(ForkJoinPool.java:2237)
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1933)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	... 47 more
Caused by: java.lang.NoSuchMethodError: 'void com.redhat.devtools.intellij.telemetry.core.service.TelemetryMessageBuilder.<init>(com.intellij.openapi.extensions.PluginDescriptor)'
	at com.redhat.devtools.intellij.quarkus.TelemetryService.lambda$new$0(TelemetryService.java:29)
	at com.redhat.devtools.intellij.telemetry.core.util.Lazy.get(Lazy.java:27)
	at com.redhat.devtools.intellij.quarkus.TelemetryService.instance(TelemetryService.java:33)
	at com.redhat.devtools.intellij.qute.lsp.QuteServer.<init>(QuteServer.java:56)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageServerFactory.createConnectionProvider(QuteLanguageServerFactory.java:27)
	at com.redhat.devtools.lsp4ij.server.definition.extension.ExtensionLanguageServerDefinition.createConnectionProvider(ExtensionLanguageServerDefinition.java:54)
	at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$0(LanguageServerWrapper.java:223)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 53 more
```

### PR DESCRIPTION
**Pull request by angelozerr**
fix: Don't crash language servers if telemetry cannot be initialized

Fixes #1384

### CODE REVIEW

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryEventName.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package com.redhat.devtools.intellij.quarkus.telemetry;
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryEventName.java:1]: header

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryService.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ */
+public interface TelemetryService {
+
+    String LSP_PREFIX = "lsp-";
+
+    String UI_PREFIX = "ui-";
+
+    String MODEL_PREFIX = "model-";
+
+    String RUN_PREFIX = "run-";
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryService.java:35]: I'd say those belong to TelemetryEventName

Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryService.java:35]: Indeed I wanted to do that, but it seems that it is not possible to declare constants in Java enum?

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+public class TelemetryManager implements Disposable {
+
+    private final Project project;
+    private TelemetryService telemetryService;
+    private LanguageServerDefinitionListener telemetryListener;
+
+    public TelemetryManager(Project project) {
+        this.project = project;
+        try {
+            Class.forName("com.redhat.devtools.intellij.telemetry.core.service.TelemetryService");
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java:44]: That call is unnecessary as the telemetry plugin is not an optional dependency, so it's guaranteed com.redhat.devtools.intellij.telemetry.core.service.TelemetryService will be present

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryEventName.java (hunk 1 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryEventName.java:14]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import com.redhat.devtools.lsp4ij.server.definition.launching.UserDefinedLanguageServerDefinition;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Helper to initialize Telemetry. A Telemetry service based on <a href="https://github.com/redhat-developer/intellij-redhat-telemetry">intellij-redhat-telemetry</a>,
+ * will be instantiated if that plugin is installed, a No-Op Telemetry stub will be used otherwise.
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java:33]: will be instantiated if an API compatible plugin is installed,

**Review by fbricon — CHANGES_REQUESTED** [review]

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryService.java (hunk 2 above)
Comment (fbricon) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryService.java:35]: yes you can, just make sure they're after the enum values. Also you should use 'static' as they're constants

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        return action(eventName, null);
+    }
+
+    @Nullable
+    public TelemetryMessageBuilder.ActionMessage action(@NotNull TelemetryEventName eventName, @Nullable Map<String, String> properties) {
+        TelemetryMessageBuilder builder = getMessageBuilder();
+        if (builder == null) {
+            return null;
+        }
+        TelemetryMessageBuilder.ActionMessage action = builder.action(eventName.getEventName());
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java:76]: if builder.action API changed, that'd crash here, error wouldn't be caught

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    @Nullable
+    private TelemetryMessageBuilder getMessageBuilder() {
+        if (hasError) {
+            return null;
+        }
+        try {
+            return builder.get();
+        }
+        catch(Exception e) {
+            hasError = true;
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java:94]: need to log error here, else we wouldn't know something's wrong

File: src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteServer.java [hunk 7]
```diff
@@ -52,12 +52,8 @@ public QuteServer(Project project) {
...
         super.setCommands(commands);
 
-        try {
-            TelemetryService.instance().action(TelemetryService.LSP_PREFIX + "startQute").send();
-        }
-        catch(Exception e) {
-            LOGGER.error("Error while consuming telemetry service", e);
-        }
+        TelemetryManager.instance()
+                .send(TelemetryEventName.LSP_START_QUTE_SERVER);
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteServer.java:56]: keep  1 line

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/dashboard/QuarkusRunDashboardCustomizer.java [hunk 8]
```diff
@@ -81,17 +83,19 @@ public boolean updatePresentation(@NotNull PresentationData presentation, @NotNu
...
                     links.put(devUILabel, new SimpleColoredComponent.BrowserLauncherTag(devUIUrl) {
                         @Override
                         public void run() {
                             // Open DevUI in a Web Browser
                             super.run();
-                            // Send event with telemetry
-                            TelemetryService.instance().action(TelemetryService.UI_PREFIX + "openDevUI").send();
+                            // Send "ui-openDevUI" telemetry event
+                            TelemetryManager.instance()
+                                    .send(TelemetryEventName.UI_OPEN_DEV_UI);
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/run/dashboard/QuarkusRunDashboardCustomizer.java:98]: keep on 1 line

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/dashboard/QuarkusRunDashboardCustomizer.java [hunk 9]
```diff
@@ -81,17 +83,19 @@ public boolean updatePresentation(@NotNull PresentationData presentation, @NotNu
                         public void run() {
                             // Open Quarkus application in a Web Browser
                             super.run();
-                            // Send event with telemetry
-                            TelemetryService.instance().action(TelemetryService.UI_PREFIX + "openApplication").send();
+                            // Send "ui-openApplication" telemetry event
+                            TelemetryManager.instance()
+                                    .send(TelemetryEventName.UI_OPEN_APPLICATION);
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/run/dashboard/QuarkusRunDashboardCustomizer.java:88]: keep on 1 line

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+            return;
+        }
+        ApplicationManager.getApplication().executeOnPooledThread(() -> {
+            try{
+                message.send();
+            } catch (Exception e) {
+                LOGGER.warn("Failed to send Telemetry data : {}", e.getMessage());
+            }
+        });
+    }
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java:110]: I think we can keep the action and asyncSend methods private, no need to expose the upstream telemetry classes

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java [hunk 11]
```diff
@@ -121,24 +122,36 @@ private void allocateLocalPort() {
...
         } else {
-            telemetry.property("tool", "not found");
+            if (telemetryMessage != null) {
+                telemetryMessage.property("tool", "not found");
+            }
         }
-        telemetry.send();
-        if (executor.getId() == DefaultDebugExecutor.EXECUTOR_ID) {
+        // Send "run-run" telemetry event
+        TelemetryManager.instance().asyncSend(telemetryMessage);
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java:152]: call   TelemetryManager.instance().send(@NotNull TelemetryEventName eventName, @Nullable Map<String, String> properties) instead

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java [hunk 12]
```diff
@@ -121,24 +122,36 @@ private void allocateLocalPort() {
...
-        telemetry.property("kind", executor.getId());
-        BuildToolDelegate toolDelegate = BuildToolDelegate.getDelegate(getModule());
+        Module module = getModule();
+
+        TelemetryMessageBuilder.ActionMessage telemetryMessage = TelemetryManager.instance()
+                        .action(TelemetryEventName.RUN_RUN);
+
+        if (telemetryMessage != null) {
+            telemetryMessage.property("kind", executor.getId());
+        }
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java:132]: just create a simple Map to collect the event properties

File: src/main/java/com/redhat/devtools/intellij/quarkus/projectWizard/QuarkusModuleBuilder.java [hunk 13]
```diff
@@ -86,15 +87,18 @@ public ModuleWizardStep modifySettingsStep(@NotNull SettingsStep settingsStep) {
...
             Module module = super.createModule(moduleModel);
             wizardContext.getUserData(QuarkusConstants.WIZARD_TOOL_KEY).processImport(module);
-            telemetry.send();
+            // Send "ui-wizard" telemetry event with no error
+            TelemetryManager.instance().asyncSend(telemetryMessage);
             return module;
         } catch (IOException | InvalidDataException | ModuleWithNameAlreadyExists | JDOMException | ConfigurationException e) {
-            telemetry.error(e).send();
+            // Send "ui-wizard" telemetry event with error
+            TelemetryManager.instance().asyncSend(telemetryMessage != null ? telemetryMessage.error(e) : null);
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/projectWizard/QuarkusModuleBuilder.java:101]: TelemetryManager.instance().send(TelemetryEventName.UI_WIZARD, e);

File: src/main/java/com/redhat/devtools/intellij/quarkus/projectWizard/QuarkusModuleBuilder.java [hunk 14]
```diff
@@ -86,15 +87,18 @@ public ModuleWizardStep modifySettingsStep(@NotNull SettingsStep settingsStep) {
...
-        TelemetryMessageBuilder.ActionMessage telemetry = TelemetryService.instance().action(TelemetryService.UI_PREFIX + "wizard");
+        TelemetryMessageBuilder.ActionMessage telemetryMessage = TelemetryManager.instance()
+                .action(TelemetryEventName.UI_WIZARD);
         try {
             processDownload();
             Module module = super.createModule(moduleModel);
             wizardContext.getUserData(QuarkusConstants.WIZARD_TOOL_KEY).processImport(module);
-            telemetry.send();
+            // Send "ui-wizard" telemetry event with no error
+            TelemetryManager.instance().asyncSend(telemetryMessage);
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/projectWizard/QuarkusModuleBuilder.java:97]: TelemetryManager.instance().send(TelemetryEventName.UI_WIZARD)

File: src/main/java/com/redhat/devtools/intellij/quarkus/lsp/QuarkusServer.java [hunk 15]
```diff
@@ -55,11 +55,8 @@ public QuarkusServer(Project project) {
         commands.add("-DrunAsync=true");
         super.setCommands(commands);
 
-        try {
-            TelemetryService.instance().action(TelemetryService.LSP_PREFIX + "start").send();
-        } catch (Exception e) {
-            LOGGER.error("Error while consuming telemetry service", e);
-        }
+        TelemetryManager.instance()
+                .send(TelemetryEventName.LSP_START_MICROPROFILE_SERVER);
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/lsp/QuarkusServer.java:59]: 1 line

File: src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusDeploymentSupport.java [hunk 16]
```diff
@@ -110,22 +111,22 @@ public static void updateClasspathWithQuarkusDeployment(Module module, ProgressI
...
                         List<VirtualFile>[] files = toolDelegate.getDeploymentFiles(module, progressIndicator);
                         LOGGER.info("Adding library to " + module.getName() + " previousHash=" + previousHash + " newHash=" + actualHash);
-                        try {
-                            TelemetryService.instance().action(TelemetryService.MODEL_PREFIX + "addLibrary").send();
-                        } catch (Exception e) {
 
-                        }
+                        // Send "model-addLibrary" telemetry event
+                        TelemetryManager.instance()
+                                .send(TelemetryEventName.MODEL_ADD_LIBRARY);
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusDeploymentSupport.java:128]: 1 line

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java (hunk 3 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java [hunk 17]
```diff
@@ -121,24 +122,29 @@ private void allocateLocalPort() {
     @Nullable
     @Override
     public RunProfileState getState(@NotNull Executor executor, @NotNull ExecutionEnvironment environment) throws ExecutionException {
-        TelemetryMessageBuilder.ActionMessage telemetry = TelemetryService.instance().action(TelemetryService.RUN_PREFIX + "run");
-        telemetry.property("kind", executor.getId());
-        BuildToolDelegate toolDelegate = BuildToolDelegate.getDelegate(getModule());
+        Module module = getModule();
+
+        @NotNull Map<String, String> properties = new HashMap<>();
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java:127]: NotNull annotation is useless

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryService.java (hunk 2 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryService.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java (hunk 4 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java (hunk 5 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java (hunk 6 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java:121]: fixed by adding a warn message (and not an error message)

File: src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteServer.java (hunk 7 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteServer.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/dashboard/QuarkusRunDashboardCustomizer.java (hunk 9 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/run/dashboard/QuarkusRunDashboardCustomizer.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java (hunk 10 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/projectWizard/QuarkusModuleBuilder.java (hunk 13 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/projectWizard/QuarkusModuleBuilder.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/projectWizard/QuarkusModuleBuilder.java (hunk 14 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/projectWizard/QuarkusModuleBuilder.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/lsp/QuarkusServer.java (hunk 15 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/lsp/QuarkusServer.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusDeploymentSupport.java (hunk 16 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusDeploymentSupport.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java (hunk 12 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java (hunk 17 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java]: fixed

**Review by fbricon — APPROVED** [review]

File: src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java [hunk 18]
```diff
@@ -121,24 +122,29 @@ private void allocateLocalPort() {
     @Nullable
     @Override
     public RunProfileState getState(@NotNull Executor executor, @NotNull ExecutionEnvironment environment) throws ExecutionException {
-        TelemetryMessageBuilder.ActionMessage telemetry = TelemetryService.instance().action(TelemetryService.RUN_PREFIX + "run");
-        telemetry.property("kind", executor.getId());
-        BuildToolDelegate toolDelegate = BuildToolDelegate.getDelegate(getModule());
+        Module module = getModule();
+
+        Map<String, String> properties = new HashMap<>();
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java:127]: could be renamed to telemetryData

Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/quarkus/run/QuarkusRunConfiguration.java]: fixed

**Comment by angelozerr** [comment]
Thanks @fbricon for your great review!

**Pull request merged**